app/gui/src/controller/searcher.rs

```diff
@@ -1235,16 +1244,63 @@ impl EditMetadataGuard {
             }),
         )
     }
+
+    fn get_saved_expression(&self) -> FallibleResult<Option<NodeEditStatus>> {
+        let module = &self.graph.graph().module;
+        let mut edit_status = None;
+        module.with_node_metadata(
+            self.node_id,
+            Box::new(|m| {
+                edit_status = m.edit_status.clone();
+            }),
+        )?;
+        Ok(edit_status)
+    }
+
+    fn revert_node_expression(&self) -> FallibleResult {
+        let edit_status = self.get_saved_expression()?;
+        match edit_status {
+            None => {
+                tracing::warn!("Tried to revert edited node, but found not edit metadata.");
+            }
+            Some(NodeEditStatus::Created) => {
+                tracing::debug!("Deleting node {} after edit reverted.", self.node_id);
+                self.graph.graph().remove_node(self.node_id)?;
+            }
+            Some(NodeEditStatus::Edited { previous_expression }) => {
+                tracing::debug!(
+                    "Reverting node {} after edit reverted. Expression to revert: {}",
+                    self.node_id,
+                    &previous_expression
+                );
+                let graph = self.graph.graph();
+                graph.set_expression(self.node_id, previous_expression)?;
+            }
+        };
+        Ok(())
+    }
 }
 
-impl Drop for EditMetadataGuard {
+impl Drop for EditGuard {
     fn drop(&mut self) {
-        self.clear_node_edit_metadata().unwrap_or_else(|e| {
-            tracing::error!(
+        if self.revert_expression.get() {
+            self.revert_node_expression().unwrap_or_else(|e| {
+                tracing::error!(
+                    "Failed to revert node edit after editing ended because of an error: {}",
+                    e
+                )
+            });
+        } else {
+            tracing::debug!("Not reverting node expression after edit.")
```

Comment:
Here the sentence is correct! It tells about reverting EXPRESSION, not the node.